arangod/Agency/Agent.cpp

```diff
@@ -882,14 +888,18 @@ trans_ret_t Agent::transact(query_t const& queries) {
 
   arangodb::consensus::index_t maxind = 0; // maximum write index
 
+  // Note that we are leading (_constituent.leading()) if and only
+  // if _constituent.leaderId == our own ID. Therefore, we do not have
+  // to use leading() or _constituent.leading() here, but can simply
+  // look at the leaderID.
   auto leader = _constituent.leaderID();
   if (leader != id()) {
     return trans_ret_t(false, leader);
   }
 
   {
     CONDITION_LOCKER(guard, _waitForCV);
-    while (_preparing) {
+    while (_preparing != 0) {
```

Comment:
minor: this line 902 is inconsistent coding style with line 970.

Reply:
Agreed. Have changed in latest commit. All explicit uses of _preparing are now eradicated (except initialization in constructor) and access is via methods.